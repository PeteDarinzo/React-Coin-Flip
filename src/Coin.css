.Coin {
  padding: 0;
  margin: 0;
  min-height: 300px;
}

.Coin-image {
  padding: 0;
  margin: 0;
  height: 300px;
}