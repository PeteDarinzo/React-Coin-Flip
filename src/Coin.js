import React from "react";
import heads from "./heads.jpg";
import tails from "./tails.jpg";
import "./Coin.css";


function Coin({ face }) {

  const getFace = () => {
    if (face === "heads") {
      return heads;
    } else if (face === "tails") {
      return tails;
    } else {
      return null;
    }
  }

  const getAlt = () => {
    if (face === "heads") {
      return "heads";
    } else if (face === "tails") {
      return "tails";
    } else {
      return "no coin";
    }
  }

  return (
    <div className="Coin">
      <img className="Coin-image" src={getFace()} alt={getAlt()}></img>
    </div>
  );
}

export default Coin;