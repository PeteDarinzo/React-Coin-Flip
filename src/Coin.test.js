import { render, screen } from '@testing-library/react';
import Coin from './Coin';

it("renders without crashing", () => {
  render(<Coin face="heads" />)
  render(<Coin face="tails" />)
  render(<Coin face={null} />)
});

it("matches the snapshot", () => {
  const { asFragment } = render(<Coin face="heads" />);
  expect(asFragment()).toMatchSnapshot();
});