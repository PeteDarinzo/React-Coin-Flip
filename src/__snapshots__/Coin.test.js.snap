// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="Coin"
  >
    <img
      alt="heads"
      class="Coin-image"
      src="heads.jpg"
    />
  </div>
</DocumentFragment>
`;
